src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -969,6 +971,21 @@ def _wrapper(func):
             return func
         return _wrapper
 
+    @staticmethod
+    def MethodMap(resource=False, status=None, version=DEFAULT_VERSION):  # noqa: N802
```

Comment:
IMO this can't be made generic as the endpoints with `@RestController.Collection` decorator (auth, rbd, prometheus) do not specify `Resource_id` attribute and hence they need a separate decorator otherwise these calls get stuck here https://github.com/ceph/ceph/blob/68236db190900a0e62ca3721e5143d04f06f604c/src/pybind/mgr/dashboard/controllers/__init__.py#L871.

Reply:
Ok, let's go with this approach then!